lib/sandthorn/event_sourced_base.rb

```diff
@@ -0,0 +1,230 @@
+module Sandthorn
+  module EventSourced
+    module Base
+
+      attr_reader :id
+      attr_reader :events
+      attr_reader :current_event_version
+      attr_reader :originating_version
+      attr_reader :aggregate_stored_serialized_object #USED?
+      attr_reader :trace_information
+
+      alias :aggregate_id :id
+
+      def base_initialize
+        @current_event_version = 0
+        @originating_version = 0
+        @events = []
+      end
+
+      def save
+        events.each do |event|
+          event[:event_data] = Sandthorn.serialize event[:event_args]
+          event[:event_args] = nil #Not send extra data over the wire
+        end
+
+        unless events.empty?
+          Sandthorn.save_events(
+            events,
+            id,
+            self.class
+          )
+          @events = []
+          @originating_version = @current_event_version
+        end
+
+        self
+      end
+
+      def ==(other)
+        other.respond_to?(:id) && id == other.id
```

Comment:
I guess this is copied code, but I think we should change that the other 'thing' has the same `version` as well.